tests/test_gc.py

```diff
@@ -38,10 +38,35 @@ def common_test(direnv_project: DirenvProject) -> None:
     assert "Executing shellHook." in out2.stderr
 
 
+def common_test_clean(direnv_project: DirenvProject) -> None:
+    testenv = str(direnv_project.dir)
+
+    out3 = run(
+        ["direnv", "exec", testenv, "hello"],
+        stderr=subprocess.PIPE,
+        check=False,
+        cwd=direnv_project.dir,
+    )
+    sys.stderr.write(out3.stderr)
+
+    files = [
+        path for path in (direnv_project.dir / ".direnv").iterdir() if path.is_file()
+    ]
+    rcs = [f for f in files if f.match("*.rc")]
+    profiles = [f for f in files if not f.match("*.rc")]
+    if len(rcs) != 1 or len(profiles) != 1:
+        print(list(files))
+    assert len(rcs) == 1
+    assert len(profiles) == 1
+
+
 def test_use_nix(direnv_project: DirenvProject) -> None:
     direnv_project.setup_envrc("use nix")
     common_test(direnv_project)
 
+    direnv_project.setup_envrc("use nix --argstr 'echo Executing hijacked shellHook.'")
```

Comment:
Shouldn't `argstr` swallow *two* arguments here? The name of the argument and the value to pass in?
I'd expec this to be:

```
direnv_project.setup_envrc("use nix --argstr shellHook 'echo Executing hijacked shellHook'")
```

I haven't tested, but I'm suspicious of why tests are passing with this (seemingly) obvious flaw.
Am I overlooking something?

Reply:
Or does the exact value of the `argstr` argument simply not matter as it is used only as a "cache buster" here?

I'd still expect `nix print-dev-env` to fail with this invocation though,as I think this will end up creating a value of `extra_args`in `use_nix` that looks like:

```bash
extra_args=("--argstr "'echo Executing hijacked shellHook.'" "")
```

Which I would expect to fail.

Reply:
Yeah that was a mistake. I assume the test didn't fail because the tests that come afterwards only check that a specific number of files are present in `.direnv/` and would be verified if the environment weren't reloaded at all (because reloading the environment itself failed).

Reply:
This indicates to me that the test itself is flawed since we don't notice failure cases. I'm not sure what to do about this (testing for things like nix-direnv is always a bit difficult due to the stateful nature of everything that is involved). Making things worse, I can't run the tests locally (unsure why but this is a longstanding issue that I have yet to figure out :/) to work on the flakes. I think the `--impure` flag should do, but I'm not sure how much I trust the test's current passing state given that there were no failures in the intermediate state

Reply:
I would argue that the test is not flawed since it wasn't noticing a failure of the test itself. The point of the test is to test that the code is correct, not that the test itself is correct. But an easy addition would be to also check the exit code of the various commands (I assume the exit code would have been non-zero with `use_nix --argstr "echo Executing hijacked shellHook."`).

Overall, assuming the tests I wrote are correct, we can be confident that `.direnv/` gets cleaned. But we can't say anything about `.direnv/flake-inputs` and `/nix/var/nix/gcroots`, so I wouldn't trust the new code too much as well.